bin/wasm-node/javascript/package.json

```
@@ -13,18 +13,26 @@
     "url": "https://github.com/paritytech/smoldot/issues"
   },
   "files": ["dist"],
-  "type": "module",
-  "types": "dist/index.d.ts",
-  "main": "dist/index.js",
+  "main": "dist/cjs/index.js",
+  "types": "dist/mjs/index.d.ts",
+  "module": "dist/mjs/index.js",
+  "exports": {
+    ".": {
+        "import": "./dist/mjs/index.js",
+        "require": "./dist/cjs/index.js"
+    }
+  },
   "scripts": {
-    "prepublishOnly": "node prepare.js --release && rimraf ./dist && tsc",
-    "build": "node prepare.js --release && rimraf ./dist && tsc",
-    "start": "node prepare.js --debug && rimraf ./dist && tsc && node demo/demo.js",
-    "test": "node prepare.js --debug && rimraf ./dist && tsc && ava --timeout=2m --concurrency 2 --no-worker-threads"
+    "buildAll": "tsc -p tsconfig-mjs.json && tsc -p tsconfig-cjs.json && bash fix-package-type",
```

```suggestion
    "build": "tsc -p tsconfig-mjs.json && tsc -p tsconfig-cjs.json && bash fix-package-type",
```

There's no reason to change the name

---

I am not sure what you mean. There is a `build` as main build command that calls this `buildAll`, which is also used many times across all scripts. I prefer to rename than to duplicate this triple command many times.

---

I see.

I'm not really happy with the name `buildAll`, but I don't have anything better coming to mind